hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -4099,6 +4099,20 @@ def __matmul__(self, other):
 
         return res if result_ndim > 0 else res[()]
 
+    @typecheck_method(axis=nullable(oneof(int, tupleof(int))))
+    def sum(self, axis=None):
+        if axis is None:
+            axis = tuple(range(self.ndim))
+
+        axis = wrap_to_tuple(axis)
+        res_ir = ir.NDArrayAgg(self._ir, axis)
+
+        num_axes_deleted = len(set(axis))
+        if num_axes_deleted > self.ndim:
+            raise ValueError(f"axes list was of length {num_axes_deleted} but ndarray dim is only {self.ndim}")
```

Comment:
I'm confused what this is guarding against. Should it just check that `0 <= i < self.ndim` for all `i in axis`?

Reply:
You're right, this just checked that you didn't specify more dimensions than the ndarray had. Replaced with better checks.